Source/Fuse.Nodes/Visual.Children.uno

```diff
@@ -219,27 +248,24 @@ namespace Fuse
 		}
 
 		/**
-			Inserts several nodes at the index. This ensures they are all added befor starting 
+			Inserts several nodes at the location. This ensures they are all added befor starting 
 			any rooting behaviouir, thus guaranteeing they are inerted in consecutive order
 			in the Children list (something that calling `Insert` in sequence cannot do, as
 			rooting a child could introduce new children).
 		*/
-		internal void InsertNodes(int index, IEnumerator<Node> items)
+		internal void InsertNodesAfter(Node preceeder, IEnumerator<Node> items)
 		{
-			InsertNodesImpl(index, items, false);
+			InsertNodesAfterImpl(preceeder, items, false);
 		}
 
-		internal void InsertOrMoveNodes(int index, IEnumerator<Node> items)
+		internal void InsertOrMoveNodesAfter(Node preceeder, IEnumerator<Node> items)
 		{
-			InsertNodesImpl(index, items, true);
+			InsertNodesAfterImpl(preceeder, items, true);
 		}
 		
 		
-		void InsertNodesImpl(int index, IEnumerator<Node> items, bool allowMove)
+		void InsertNodesAfterImpl(Node preceeder, IEnumerator<Node> items, bool allowMove)
 		{
-			if (index <0 || index > Children.Count)
```

Comment:
Perhaps replace this sanity check with one that checks that `preceeder` is a child of this node.  It's zero cost now I think, and it'd give a cleaner error than waiting until deep inside the loop.